deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/BaseDeepLinkDelegate.java

```diff
@@ -210,15 +215,34 @@ public DeepLinkResult dispatchFrom(Activity activity, Intent sourceIntent) {
 
   private DeepLinkEntry findEntry(String uriString) {
     DeepLinkEntry entryRegExpMatch = null;
-    DeepLinkEntry entryIdxMatch;
-    DeepLinkUri parse = DeepLinkUri.parse(uriString);
+    List<DeepLinkEntry> entryIdxMatches = new ArrayList<>();
+    DeepLinkUri uri = DeepLinkUri.parse(uriString);
     for (BaseRegistry registry : registries) {
-      entryIdxMatch = registry.idxMatch(parse, configurablePathSegmentReplacements);
+      DeepLinkEntry entryIdxMatch = registry.idxMatch(uri, configurablePathSegmentReplacements);
       if (entryIdxMatch != null) {
-        return entryIdxMatch;
+        entryIdxMatches.add(entryIdxMatch);
+      }
+    }
+    // Found no match
+    if (entryIdxMatches.isEmpty()) {
+      return null;
+    } else if (entryIdxMatches.size() == 1) {
+      // Found just one match
+      return entryIdxMatches.get(0);
+    }
+    // Found multiple matches. Sort matches by concreteness:
+    // No variable element > containing placeholders >  are a configurable path segment
+    Collections.sort(entryIdxMatches, new Comparator<DeepLinkEntry>() {
+      @Override
+      public int compare(DeepLinkEntry a, DeepLinkEntry b) {
+        return a.moreConcreteThan(b);
       }
+    });
+    if (entryIdxMatches.get(0).moreConcreteThan(entryIdxMatches.get(1)) == 0) {
+      Log.w(TAG, "More than one match with the same concreteness!! ("
```

Comment:
I guess this is a bit indeterministic which gets called. Would be nice to be able to hook into this warning log and crash in development (maybe could add optional constructor injected logger interface)

Reply:
Yeah, let me add that.

Reply:
@BenSchwab I added that and a test for it. Let me know if that is what you ad in mind?